internal/tiger/cmd/db.go

```diff
@@ -932,3 +960,363 @@ func isConnectionRejected(err error) bool {
+	service api.Service,
+	role string,
+) error {
+	_, err := resetServicePassword(ctx, cfg, service, role, cmd.ErrOrStderr())
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cmd.ErrOrStderr(), "To view your new password, run: tiger service get %s --with-password\n", util.Deref(service.ServiceId))
+
+	// Launch psql (password is now in storage, launchPsql will retrieve it)
+	return launchPsql(details.String(), psqlPath, psqlFlags, service, role, cmd)
+}
+
+// connectWithPasswordMenu handles the connection flow if the stored password is invalid
+// Offers an interactive menu to enter the password manually or reset it
+func connectWithPasswordMenu(
+	ctx context.Context,
+	cmd *cobra.Command,
+	service api.Service,
+	details *common.ConnectionDetails,
+	psqlPath string,
+	psqlFlags []string,
+	role string,
+	opts connectOptions,
+) error {
+	// Load config (needed for password reset)
+	cfg, err := config.Load()
+	if err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+
+	// Handle --reset-password flag (headless mode)
+	if opts.resetPassword {
+		if err := resetPasswordAndLaunchPsql(ctx, cmd, cfg, details, psqlPath, psqlFlags, service, role); err != nil {
+			return fmt.Errorf("failed to reset password: %w", err)
+		}
+		return nil
+	}
+
+	// Handle --password flag (headless mode)
+	if opts.password != "" {
+		err := testSaveAndLaunchPsqlWithPassword(ctx, cmd, details, psqlPath, psqlFlags, service, opts.password, role)
+		if err != nil {
+			if isAuthenticationError(err) {
+				return fmt.Errorf("authentication failed: invalid password")
+			}
+			return fmt.Errorf("connection failed: %w", err)
+		}
+		return nil
+	}
+
+	// Interactive mode: Get stored password (if any)
+	storage := common.GetPasswordStorage()
+	storedPassword, _ := storage.Get(service, role)
+
+	// Try to connect with stored password first
+	err = testConnectionWithPassword(ctx, details, storedPassword, 10*time.Second)
+	if err == nil {
+		// Password works, launch psql
+		return launchPsql(details.String(), psqlPath, psqlFlags, service, role, cmd)
+	}
+
+	// Check if it's an auth error
+	if !isAuthenticationError(err) {
+		// Non-auth error (network, timeout, etc.) - report it directly
+		return fmt.Errorf("connection failed: %w", err)
+	}
+	// Auth failed with stored password, continue to recovery menu
+	fmt.Fprintf(cmd.ErrOrStderr(), "Stored password is invalid or expired.\n\n")
+
+	// Check if we're in a TTY for interactive menu
+	if !checkStdinIsTTY() {
+		return fmt.Errorf("authentication failed and no TTY available for interactive password entry. " +
+			"Use --password or --reset-password flag for headless operation")
+	}
+
+	// Interactive recovery loop
+	for {
+		option, err := selectPasswordRecoveryOption(cmd.ErrOrStderr())
+		if err != nil {
+			return err
+		}
+
+		switch option {
+		case optionEnterPassword:
+			// Prompt for password
+			fmt.Fprint(cmd.ErrOrStderr(), "Enter password: ")
+			password, err := readPasswordFromTerminal()
```

Comment:
We should wrap any reads from `stdin` in the [readString](https://github.com/timescale/tiger-cli/blob/b936b4a160a1914f50feb9b2ff96179f942a694d/internal/tiger/cmd/root.go#L174-L200) function, which correctly handles context cancellation (i.e. `ctrl-c`). Otherwise, users won't be able to exit the program with `ctrl-c` while this prompt is shown. See how its done [here](https://github.com/timescale/tiger-cli/blob/b936b4a160a1914f50feb9b2ff96179f942a694d/internal/tiger/cmd/db.go#L363), for example.

The one exception to that rule is reads from stdin that go through bubbletea, since that provides its own `ctrl-c` handling.